safe_autonomy_simulation/dynamics/dynamics.py

```diff
@@ -56,12 +64,22 @@ def step(
         Tuple[np.ndarray, np.ndarray]
             Tuple of the system's next state and the state's instantaneous time derivative at the end of the step
         """
+        state = cast_jax(state, use_jax=self.use_jax)
+        control = cast_jax(control, use_jax=self.use_jax)
         next_state, state_dot = self._step(step_size, state, control)
-        next_state = self.np.clip(next_state, self.state_min, self.state_max)
+
+        next_state = self.np.clip(
```

Comment:
I was playing around with jax and it looks like the jnp `clip` is slower than the numpy version, even if you use jit.  So, we can probably just use `numpy.clip` all the time.  Not sure if this provides a simplification that will allow us to localize conversion between jax and numpy to a smaller area in the code base

Reply:
If its faster than we can just use numpy here! The state limits might be jax arrays but it seems numpy can handle mixing types just fine. I've committed a change to just use `np.clip` here.